exonum/src/proto/binary_map.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::proto;
+use exonum_crypto::Hash;
+use exonum_merkledb::{BinaryValue, ObjectHash};
+use exonum_proto::ProtobufConvert;
+use protobuf::Message;
+
+use std::{borrow::Cow, collections::BTreeMap};
+
+/// Protobuf wrapper type to store small maps of non-scalar keys and values.
+#[derive(Debug, Default, Clone, Eq, PartialEq, Ord, PartialOrd, Serialize, Deserialize)]
+pub struct BinaryMap<K: Ord, V>(pub BTreeMap<K, V>);
```

Comment:
Just to be clear: Protobuf's `map<string, bytes>` is not used because it has no guarantees as to the ordering of keys (i.e., maps to Rust's `HashMap`), right?

Reply:
Right, as protobuf [documentation](https://developers.google.com/protocol-buffers/docs/proto3#maps) said:
```
Wire format ordering and map iteration ordering of map values is undefined, so you 
cannot rely on your map items being in a particular order
```

Reply:
May you write note about key's ordering and policy about the duplicate entries?